apps/arena/lib/arena/game_updater.ex

```diff
@@ -472,16 +473,30 @@ defmodule Arena.GameUpdater do
   # End game flow
   ##########################
 
-  defp apply_zone_damage(players, zone) do
+  defp apply_zone_damage(players, zone, zone_damage_interval) do
     safe_zone = Entities.make_circular_area(0, %{x: 0.0, y: 0.0}, zone.radius)
     safe_ids = Physics.check_collisions(safe_zone, players)
     to_damage_ids = Map.keys(players) -- safe_ids
+    now = System.monotonic_time(:millisecond)
 
     Enum.reduce(to_damage_ids, players, fn player_id, players_acc ->
-      Player.change_health(players_acc, player_id, 1)
+      player = Map.get(players_acc, player_id)
+      last_damage = player |> get_in([:aditional_info, :last_damage_received])
+      elapse_time = now - last_damage
+
+      player = player |> maybe_receive_zone_damage(elapse_time, zone_damage_interval)
+
+      Map.put(players_acc, player_id, player)
     end)
   end
 
+  defp maybe_receive_zone_damage(player, elapse_time, zone_damage_interval)
+       when elapse_time > zone_damage_interval do
+    Player.change_health(player, 1)
```

Comment:
@BertovDev this 1 should be a config, because with this configuration the zone needs 100 seconds to kill a player and that doesn't make sense

Reply:
I agree with you. Will add it